### PR TITLE
adding support for integration testing of SSE with other features

### PR DESCRIPTION
**Pull request by hmaheswa**
integration testing of server-side-encryption(s3 and kms) with transition, bucket notifications, resharding, sts and bucket policy.

polarion id: https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83586489

bz for md5 mismatch after transition : https://bugzilla.redhat.com/show_bug.cgi?id=2262741
bz for sse copy failure: https://bugzilla.redhat.com/show_bug.cgi?id=2149450

pass logs: http://magna002.ceph.redhat.com/ceph-qe-logs/Hemanth_Sai/sse_integration_testing_PR/

fail logs:
http://magna002.ceph.redhat.com/ceph-qe-logs/Hemanth_Sai/sse_kms_gklm/test_sse_s3_per_bucket_with_notifications_dynamic_reshard.console.log
http://magna002.ceph.redhat.com/ceph-qe-logs/Hemanth_Sai/sse_integration_testing_PR/test_sse_kms_per_bucket_multipart_object_download_after_transition.console.log

### CODE REVIEW

**Review by viduship — COMMENTED** [review]
@hmaheswa , Please provide the test case IDs if any for this PR

File: rgw/v2/tests/s3_swift/reusable.py [hunk 1]
```diff
@@ -194,11 +194,22 @@ def upload_object(
...
         }
     )
+
+    args = [s3_object_path]
+    if config.test_ops.get("sse_s3_per_object") is True:
+        if config.encryption_keys == "s3":
+            log.info("SSE S3 AES256 encryption method applied")
+            extra_args = {"ServerSideEncryption": "AES256"}
+            args.append(extra_args)
+        elif config.encryption_keys == "kms":
```
Comment (viduship) [inline rgw/v2/tests/s3_swift/reusable.py:204]: Are we testing with KMS also?

File: rgw/v2/tests/s3_swift/test_bucket_notifications.py [hunk 2]
```diff
@@ -238,6 +240,21 @@ def test_exec(config, ssh_con):
                 if config.test_ops.get("verify_persistence_with_kafka_stop", False):
                     notification.start_stop_kafka_server("stop")
 
+                # Choose the encryption_method sse-s3 or sse-kms
```
Comment (viduship) [inline rgw/v2/tests/s3_swift/test_bucket_notifications.py:243]: @hmaheswa Looks these lines are reusable. Please put them in a method in reusable.py and call

Comment (hmaheswa) [reply rgw/v2/tests/s3_swift/test_bucket_notifications.py]: done @viduship 
please re-review

**Review by viduship — DISMISSED** [review]

**Review by ckulal — COMMENTED** [review]
looks good

File: rgw/v2/tests/s3_swift/configs/test_sse_s3_per_object_with_sts.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# polarion test case id: CEPH-83586489
+# test scripts : test_sts_using_boto.py
+config:
```
Comment (anrao19) [inline rgw/v2/tests/s3_swift/configs/test_sse_s3_per_object_with_sts.yaml:3]: Please update this config in script file mentioned here

Comment (hmaheswa) [reply rgw/v2/tests/s3_swift/configs/test_sse_s3_per_object_with_sts.yaml:3]: done @anrao19

**Review by mkasturi18 — APPROVED** [review]

**Pull request merged**